File: apps/api/app/models/session.py
"""Pydantic models for session data."""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class MeetingResponse(BaseModel):
    """Meeting (race weekend) response model."""

    meeting_key: int
    meeting_name: str
    location: str
    official_name: str
    country_key: int
    country_name: str
    circuit_key: int
    circuit_short_name: str
    date_start: str
    date_end: str
    year: int
    gmt_offset: Optional[str] = None


class SessionResponse(BaseModel):
    """Session response model."""

    session_key: int
    meeting_key: int
    location: str
    session_type: str = Field(..., description="practice, qualifying, sprint, or race")
    session_name: str
    date_start: str
    date_end: str
    country_name: str
    circuit_short_name: str
    year: int
    gmt_offset: Optional[str] = None

File: apps/web/src/types/session.ts
export interface Session {
  session_key: number;
  meeting_key: number;
  location: string;
  session_type: 'practice' | 'qualifying' | 'sprint' | 'race';
  session_name: string;
  date_start: string;
  date_end: string;
  country_name: string;
  circuit_short_name: string;
  year: number;
}

export interface Meeting {
  meeting_key: number;
  meeting_name: string;
  location: string;
  official_name: string;
  country_key: number;
  country_name: string;
  circuit_key: number;
  circuit_short_name: string;
  date_start: string;
  date_end: string;
  year: number;
}
